packages/core/mesh/teleport/src/muxing/framer.ts

```diff
@@ -82,6 +98,33 @@ export class Framer {
     responseQueue.forEach((cb) => cb());
   }
 
+  private async _processChunkQueue() {
+    if (this._chunkQueue.length === 0) {
+      return;
+    }
+
+    const chunk = this._chunkQueue.shift()!;
+    await this._sendData(chunk.msg);
+    chunk.callback?.();
+
+    await this._processChunkQueue();
+  }
+
+  private async _sendData(data: Buffer) {
+    return new Promise<void>((resolve) => {
+      const canContinue = this._stream.push(data);
+      if (!canContinue) {
+        this._responseQueue.push(resolve);
+      } else {
+        process.nextTick(resolve);
```

Comment:
is nextTick really needed here? I though resolving the promise would create a microtask anyway